### PR TITLE
Setting up instance stats so that we could connect to our agent

### PR DESCRIPTION
**Pull request by dvargas92495**
Surfacing enough info so that we could SSH into the box and start running our onboarding script. Still todo:

- Showing the SSH info in the UI
- Confirm successful connection to the box
- Run onboarding script in box

Intentionally deferring:
- Polling status on employee page
- Defining responsibilities (next)
- Automatically running onboarding script on the box

### CODE REVIEW

**Comment by dvargas92495** [comment]
Great progress! https://www.loom.com/share/8c6a80820a994821b5c6a9ff63590904

Gonna merge through and shift focus to actual runtime stuff now

File: app/data/createUserEmployee.server.ts [hunk 1]
```diff
@@ -36,24 +37,58 @@ const createUserEmployee = async ({
...
+    throw new InternalServerError("Failed to create key pair");
+  }
+
   const instance = await ec2.runInstances({
     ImageId: UBUNTU_SERVER_22_04_LTS,
+    KeyName: keyPair.KeyName,
+    UserData: Buffer.from(
+      `\
+!#/bin/bash
+curl https://samepage.network/scripts/bootstrap.sh | bash
```
Comment (dvargas92495) [inline app/data/createUserEmployee.server.ts:56]: this didn't actually run

File: app/routes/__private/user/employees.$uuid.tsx [hunk 2]
```diff
@@ -9,16 +9,35 @@ import remixAppAction from "~/data/remixAppAction.server";
...
-      <h3 className="text-md italic mb-16">{employee.instanceId}</h3>
+      <div className="mb-16 border-dashed border p-4">
+        <h3 className="text-lg font-bold mb-4">Employee Device Info</h3>
+        <p className="italic mb-2">
+          {employee.instance.id} - {employee.instance.state}
+        </p>
+        <div className="flex justify-between items-start">
+          <code>
+            ssh -i {employee.uuid}.pem {employee.instance.username}@
+            {employee.instance.dnsName}
```
Comment (dvargas92495) [inline app/routes/__private/user/employees.$uuid.tsx:26]: I needed to also `chmod 400 {file}.pem` - annoying

File: app/routes/__private/user/employees.$uuid.tsx [hunk 3]
```diff
@@ -9,16 +9,35 @@ import remixAppAction from "~/data/remixAppAction.server";
...
+            reloadDocument
+          >
+            Download Private Key
+          </Link>
+        </div>
+      </div>
       <div className="flex-grow">
         <h2 className="text-2xl">Responsibilities</h2>
-        {responsibilities.map((r) => (
+        {employee.responsibilities.map((r) => (
```
Comment (dvargas92495) [inline app/routes/__private/user/employees.$uuid.tsx:40]: deferring this for now.

Devin has a notion of "Tips" which is kind of cool. However, I realized through that experience that I didn't want to enter it in Devin's UI

File: app/routes/__private/user/employees.$uuid.tsx [hunk 4]
```diff
@@ -56,15 +75,12 @@ export const loader: LoaderFunction = (args) => {
 
 export const action: ActionFunction = (args) => {
   return remixAppAction(args, {
-    DELETE: ({ userId, params, requestId, ...rest }) => {
-      console.log("params", params);
-      console.log("rest", rest);
-      return fireUserEmployee({
+    DELETE: ({ userId, params, requestId }) =>
+      fireUserEmployee({
```
Comment (dvargas92495) [inline app/routes/__private/user/employees.$uuid.tsx:79]: need to still delete the key here, and add a history item record for the employee

**Review by mdroidian — COMMENTED** [review]
🔥

**Pull request merged**